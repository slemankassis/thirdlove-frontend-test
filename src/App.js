import React from 'react';
import './App.css';
import Product from './pages/Product';
// import hooks from './hooks';
import mockData from './mockData';

function App() {
  // const [data, loading] = hooks.useFetch(
  //   'https://www.mocky.io/v2/5c6c3a92320000e83bbef971',
  // );

  // Test with mocked data
  const [data, loading] = [mockData, false];

  // TODO: Use ErrorBoundary here for catch errors in Product component
  return (
    <div>
      {loading
        ? 'Loading...'
        : <Product product={data.product} />}
    </div>
  );
}

export default App;
