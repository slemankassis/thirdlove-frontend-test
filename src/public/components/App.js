const React = require('react');
const { useFetch } = require('../../hooks');
const ProductView = require('./ProductView');

// eslint-disable-next-line react/prop-types
const App = ({ pageTitle }) => {
  const [data, loading] = useFetch(
    'https://www.mocky.io/v2/5c6c3a92320000e83bbef971',
  );

  return (
    <div>
      {loading ? (
        `Loading... ${pageTitle}`
      ) : (
        <ProductView product={data.product} />
      )}
    </div>
  );
};

module.exports = App;
