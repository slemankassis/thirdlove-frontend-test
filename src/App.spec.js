import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

describe('App', () => {
  it('deep rendering without crashing using ReactDOM', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders snapshot without info', () => {
    const component = shallow(<App />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
