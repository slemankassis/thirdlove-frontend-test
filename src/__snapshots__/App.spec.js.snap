// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snaps renders snapshot without info 1`] = `
"<div>
  Loading...
</div>"
`;
